src/test/java/org/commcare/formplayer/tests/MultiSelectCaseListTest.java

```diff
@@ -65,6 +68,19 @@ public void testNormalMultiSelectCaseList() throws Exception {
         checkForSelectedCasesInstance(formRespUsingGuid.getSessionId(), selections);
     }
 
+    @Test
+    public void testMultiSelectCaseListWithPreValidatedInput() throws Exception {
+        Mockito.doReturn(true)
+                .when(restoreFactoryMock).isConfirmedSelection(any());
+        String[] selections = new String[]{"0", "1", "use_selected_values"};
+        String[] selectedValues =
+                new String[]{"5e421eb8bf414e03b4871195b869d894", "3512eb7c-7a58-4a95-beda-205eb0d7f163"};
+        NewFormResponse formResp = sessionNavigateWithSelectedValues(selections, APP, selectedValues,
+                NewFormResponse.class);
+        sessionNavigateWithSelectedValues(selections, APP, selectedValues,
```

Comment:
how did this fail before? It's not clear from the test what's actually being tested.

Reply:
it was hitting a session navigation exception due to the datum being set as  `use_selected_values` [here](https://github.com/dimagi/formplayer/blob/fixesForMultiSelect/src/main/java/org/commcare/formplayer/session/MenuSession.java#L176) . Probably [this](https://github.com/dimagi/formplayer/pull/1122/commits/4b585d8cb76a493b00f23e00cdf37b6f7a52856e) is better failure mode.